### PR TITLE
EPROD-468 evm block extractor

### PR DESCRIPTION
**Pull request by veeso**
# Create a tool to extract block data from EVM canister

Issue ticket link: https://infinityswap.atlassian.net/browse/EPROD-468

### Code conventions

- [x] I have performed a self-review of my code
- [x] Every new function is documented
- [x] Object names are auto explicative

### Security

- [x] The PR does not break APIs backward compatibility
- [x] The PR does not break the stable storage backward compatibility

### Testing

- [x] Every function is properly unit tested
- [ ] I have added integration tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/evm-block-extractor/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    #[arg(long = "rpc-url", short('u'))]
+    rpc_url: String,
+
+    /// Output ZIP file to write blocks to
+    #[arg(long = "output", short('o'))]
+    output_file: PathBuf,
+
+    /// block to start with
+    #[arg(long, short('s'), default_value = "0x0")]
+    start_block: String,
```
Comment (ufoscout) [inline src/evm-block-extractor/src/main.rs:33]: this could be number

File: src/evm-block-extractor/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    // Initialize logger
+    init_logger()?;
+    let args = Args::parse();
+
+    let start_block = u64::from_str_radix(&args.start_block.replace("0x", ""), 16)?;
+    let end_block = args
```
Comment (ufoscout) [inline src/evm-block-extractor/src/main.rs:47]: they should be numbers, there's no need for this convertion

File: src/evm-block-extractor/src/rpc_client.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+use anyhow::Context;
+use ethers_core::types::{Block, BlockNumber, Transaction};
+use serde_json::{json, Value};
+
+/// Get Blocks by number
+pub async fn get_blocks_by_number(
```
Comment (ufoscout) [inline src/evm-block-extractor/src/rpc_client.rs:6]: We decided to collect the receipts too

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: src/evm-block-extractor/src/main.rs (hunk 1 above)
Comment (veeso) [reply src/evm-block-extractor/src/main.rs:33]: I think hex is more used for blocks, like for the API

Comment (veeso) [reply src/evm-block-extractor/src/main.rs:33]: we used string also for the benchmark tool

**Comment by veeso** [comment]
@ufoscout changed from hex to base 10

**Review by ufoscout — APPROVED** [review]

**Pull request merged**